docs/roi_tables.md

```diff
@@ -0,0 +1,333 @@
+  object with a `path` key and a string value (i.e. the path to the data the
+  table is annotating).
+* Attributes may include a key `instance_key`, which is the key in `obs` that
+  denotes which instance in `region` the row corresponds to. If `instance_key`
+  is not provided, the values from the `_index` Zarr attribute of `obs` is used.
+
+Here is an example of `image.zarr/tables/table1/.zattrs`
+```json
+{
+    "fractal_roi_table_version": "1",
+    "type": "ngff:region_table",
+    "region": {
+        "path": "../labels/label_DAPI",
+    },
+    "instance_key": "label",
+    "encoding-type": "anndata",      # Automatically added by anndata
+    "encoding-version": "0.1.0",     # Automatically added by anndata
+}
+```
+
+### AnnData table format
+
+Data of a table are stored into a Zarr group as AnnData ("Annotated Data")
+objects; the [`anndata` Python library](https://anndata.readthedocs.io) provides the
+definition of this format and the relevant tools.
+
+Quoting from `anndata` documentation:
+
+> AnnData is specifically designed for matrix-like data. By this we mean that
+> we have $n$ observations, each of which can be represented as $d$-dimensional
+> vectors, where each dimension corresponds to a variable or feature. Both the
+> rows and columns of this $n \times d$ matrix are special in the sense that
+> they are indexed.
+>
+> (https://anndata.readthedocs.io/en/latest/tutorials/notebooks/getting-started.html)
+
+Note that AnnData tables are easily transformed from/into `pandas.DataFrame`
+objects - see e.g. the [`AnnData.to_df`
+method](https://anndata.readthedocs.io/en/latest/generated/anndata.AnnData.to_df.html#anndata.AnnData.to_df).
+
+### Table contents
+
+The [`var` attribute of AnnData
+objects](https://anndata.readthedocs.io/en/latest/generated/anndata.AnnData.var.html#anndata.AnnData.var)
+indexes the columns of the table. A `fractal-tasks-core` ROI table must include
+the following six columns:
+
+* `x_micrometer`, `y_micrometer`: the lower bounds of the XY intervals defining the ROI, in micrometers;
+* `z_micrometer`: the lower bound of the Z interval defining the ROI, in arbitrary units or in micrometers;
+* `len_x_micrometer`, `len_y_micrometer`: the XY edge lenghts, in micrometers;
+* `len_z_micrometer`: the Z edge lenght in arbitrary units (corresponding to the number of Z planes) or in micrometers.
+
+ROI tables may also include other optional columns:
+
+* `x_micrometer_original` and `y_micrometer_original`, which are a copy of `x_micrometer` and `y_micrometer` taken before applying some transformation;
+* `label`, which is used within measurement tables as a reference to the labels corresponding to a row of measurements (see [description of `instance_key` above](#single-table-segmented-objects)).
+
+> Notes:
+>
+> 1. The **axes origin** for the ROI positions (e.g. for `x_micrometer`) is set
+>    to coincide with the top-left corner of a well (for the YX axes) and with
```

Comment:
Top left corner of an image
=> Our spec should apply to any OME-NGFF image, not only to plates with wells